src/redset.h

```diff
@@ -40,10 +60,37 @@ typedef void* redset_filelist;
 ///@{
 
 /** initialize library */
-int redset_init();
+int redset_init(void);
 
 /** shutdown library */
-int redset_finalize();
+int redset_finalize(void);
+
+/* needs to be above doxygen comment to get association right */
+typedef struct kvtree_struct kvtree;
+/**
+ * Get/set redset configuration values.
+ *
+ * The following configuration options can be set (type in parenthesis):
+ *   * "DEBUG" (int) - if non-zero, output debug information from inside
+ *     redset.
+ *   * "SETSIZE" (int) - set size for redset to use.
+ *   * "MPI_BUF_SIZE" (byte count [IN], int [OUT]) - MPI buffer size to chunk
+ *     file transfer. Must not exceed INT_MAX.
+ *   .
+ * Symbolic names REDSET_KEY_CONFIG_FOO are defined in redset.h and should
+ * be used instead of the strings whenever possible to guard against typos in
+ * strings.
+ *
+ * \result If config != NULL, then return config on success.  If config=NULL
+ *         (you're querying the config) then return a new kvtree on success.
+ *         NULL on any failures.
+ * \param config The new configuration. If config=NULL, then return a kvtree
+ *                with all the configuration values.
```

Comment:
I like how you mentioned that the caller is responsible for freeing any returned kvtree in other components.  Let's add that here, too.

Reply:
Added statement that returned kvtree must be `kvtree_delete()`ed by the caller in f63e2ea